Add tour to ccdb

flacoman91 opened the pull request:
Short description explaining the high-level reason for the pull request

## Additions

-

## Removals

-

## Changes

-

## Testing

-

## Screenshots


## Notes

-

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

flacoman91 commented:
@anselmbradford this is almost ready.  I still have some things I need to work out in the Tour component.

Could you take over and assist with the styling?

flacoman91 commented on src/actions/trends.jsx:
```diff
@@ -107,6 +108,21 @@ export function removeFocus() {
+ * @param {string} value the new payload from the tooltip
+ * @returns {string} a packaged payload to be used by Redux reducers
+ */
+export function tourToggled( value ) {
```
I would write a test for this, but it's unnecessary esp if we are going to convert to redux toolkit.

anselmbradford commented on src/components/Tour/Tour.jsx:
```diff
@@ -0,0 +1,111 @@
+
+    // hacks to setup the page. we should figure out if we should
+    // have this in redux instead
+    if ( expand === 9 ) {
```
Should the `9` go into a constant?

flacoman91 replied:
I updated this so it's more scalable so we don't have to hardcode the number.

anselmbradford commented on src/components/Tour/Tour.jsx:
```diff
@@ -0,0 +1,111 @@
+      document.querySelector( '.advanced-container button' ).click()
+    }
+
+    if ( expand === 16 ) {
```
Should the `16` go into a constant?

flacoman91 replied:
same as above.

anselmbradford commented on src/components/Tour/Tour.jsx:
```diff
@@ -0,0 +1,111 @@
+      // if not equal, it means one is open so we don't need to expand a tick
+      // i would prefer to collapse all of the bars and expand only the first.
+      // this way you can actually highlight the "View more about XYZ Product"
+      const expandables = document.querySelectorAll( '#row-chart-product .y-axis-group .tick.expandable' );
+      const ticks = document.querySelectorAll( '#row-chart-product .y-axis-group .tick' );
```
You could eliminate the multiple row-chart-product ID lookups:

```js

const rowChartProduct = document.querySelector( '#row-chart-product' );
const expandables = rowChartProduct.querySelectorAll( '.y-axis-group .tick.expandable' );
const ticks = rowChartProduct.querySelectorAll( '.y-axis-group .tick' );

```

anselmbradford replied:
Also, if there's only one `y-axis-group`, you could include that in the first query.

flacoman91 replied:
i moved this logic into redux so the page state gets set to a good happy path.  We don't need to do this check anymore.

anselmbradford commented on src/components/Tour/TourButton.less:
```diff
@@ -0,0 +1,11 @@
+@import (less) "../../css/base.less";
```
This looks like it could be `@import (reference) "../../css/base.less";`

flacoman91 replied:
this one looks okay.

flacoman91 commented on src/components/Tour/Tour.less:
```diff
@@ -0,0 +1,16 @@
+// Intro.js styles
+@import (css) '../../../node_modules/intro.js/introjs.css';
```
No.  This may be part of the technical debt regarding our build tools. 

<img width="1117" alt="Screen Shot 2022-08-22 at 6 08 37 PM" src="https://user-images.githubusercontent.com/1038306/186027413-7930c331-d5d4-4a0c-b134-1b22e3949703.png">

flacoman91 commented:
Looks like  the cookie test  all of a sudden failing is a  known  issue

https://github.com/salesforce/tough-cookie/issues/246

anselmbradford commented:
I'm getting an error on the end tour step

 
<img width="762" alt="Screen Shot 2022-09-01 at 12 29 58 PM" src="https://user-images.githubusercontent.com/704760/187965775-0240a3cd-2599-4361-aa7a-217d36547c1e.png">

flacoman91 commented:
updates responding to GHE  2104 have been  made.

flacoman91 commented:
removed Date Range steps in trends and map as requested.

flacoman91 commented:
Restored links and table of contents. made only "Index" a title heading

<img width="1257" alt="Screen Shot 2022-09-07 at 3 39 36 PM" src="https://user-images.githubusercontent.com/1038306/188963969-ef3f1ee1-dc7f-4a90-986a-280fd943ed93.png">
<img width="1230" alt="Screen Shot 2022-09-07 at 3 39 04 PM" src="https://user-images.githubusercontent.com/1038306/188963973-d0ae2cd2-7b89-49f2-9225-5764d17faa5e.png">

anselmbradford commented on src/components/RefineBar/Select.jsx:
```diff
@@ -48,7 +48,7 @@ export class Select extends React.Component {
 
     return (
-      <section className="cf-select">
+      <section className={'cf-select ' + this.props.id}>
```
Should `this.props.id` go in an `id` attribute instead of class?

anselmbradford replied:
Ah I see in the snapshot it gets repeated. Well, instead of super generic class names like `size` and `sort`, let's prefix this to be `<section className={'cf-select cf-select__' + this.props.id}>`

flacoman91 replied:
I had to add this here so that I could target these for highlighting in intro.js here

https://github.com/cfpb/ccdb5-ui/pull/409/files#diff-1f017f3f6619b22aabaa7855af7aa91b94892609d8f7f765b5d5c942c4565d98R60

I think it might be possible to update the document.querySelector call to look for attributes instead

anselmbradford replied:
Hm yea, that actually would be the best way. The prefixed class sets this up as a modifier on `cf-select`, which implies that `cf-select__sort` (or whatever) is modifying `cf-select`. In actuality it is being used as an ID. That's not a big deal, but it could send someone off to look for the modifying CSS. If you want, you should be able to do `<section className="cf-select" data-tour={this.props.id}>` and query it with `document.querySelector('[data-tour="sort"]')`, or `document.querySelector('section[data-tour="sort"]')`.

anselmbradford commented on package.json:
```diff
@@ -149,5 +149,9 @@
     "access": "public",
     "registry": "https://registry.npmjs.org/"
+  },
+  "dependencies": {
```
I think we want everything on `devDependencies`, as I think `dependencies` end up in the npm package. Since the npm package is just delivering a built JS file, I don't think we want anything else there.

anselmbradford commented on src/components/Tour/constants/tourStepsConstants.js:
```diff
@@ -0,0 +1,370 @@
+    {
+      element: GEO.STEP_1.selector,
+      intro:
+         '<div>' +
```
trivial: indented a little too much.

anselmbradford reviewed (approved):
You could update the data- attributes, but otherwise looks good

flacoman91 commented on src/components/RefineBar/Select.jsx:
```diff
@@ -44,19 +44,21 @@ export class Select extends React.Component {
 
   render() {
-    const id = 'choose-' + this.props.id
+    const id = 'select-' + this.props.id
```
i just renamed this to select vs choose, since that makes more sense.

flacoman91 commented:
> You could update to use data- attributes, but otherwise looks good

I updated it [here](https://github.com/cfpb/ccdb5-ui/pull/409/files#diff-1ea787e5c97d51c9f0a9bdf61d2fc333eeea3f1dad7394c10c1dc8bd61f1d568R52) and also renamed the id to make it clearer.
